src/hotspot/share/cds/filemap.cpp

```diff
@@ -1083,24 +1083,29 @@ class FileHeaderHelper {
     return &_header;
   }
 
-  bool read_base_archive_name(char** target) {
+  char* read_base_archive_name() {
     assert(_fd != -1, "Archive should be open");
-    size_t name_size = (size_t)_header._base_archive_name_size;
+    size_t name_size = _header._base_archive_name_size;
```

Comment:
Unrelated, but do we do sanity checking of header data? E.g. for a max reasonable length of _base_archive_name_size?

Reply:
I am planning to do that in a follow-on RFE (https://bugs.openjdk.java.net/browse/JDK-8276769). I will add more structural checks:

- GenericCDSFileMapHeader::_header_size must be smaller than the archive file
- _base_archive_name_size/_base_archive_path_offset must be inside the header
- the header must not overlap with the regions
- the regions should not overlap with each other

Since more checks are added, I need to make sure the new checks also respect the -Xshare:auto flag (failure should not cause the VM to exit) which is the purpose of JDK-8276769.

Reply:
Sounds good.